pages/assessment.tsx

```diff
@@ -15,13 +15,33 @@ import { Button } from "@mui/material";
 import Header from "../components/header";
 import findIndex from "lodash/findIndex";
 import cloneDeep from "lodash/cloneDeep";
+import { useRouter } from "next/router";
+import { useEffect } from "react";
 
 const Assessment: NextPage = () => {
-  const taskNumber: number = 2; // Later will get it another way. And then make sure tasknumber doesn't get too high.
-  const taskTitle: string =
-    data.ext_inspera_candidates[0].result.ext_inspera_questions[taskNumber - 1]
-      .ext_inspera_questionTitle;
+  // create router object
+  const router = useRouter();
 
+  // isReady: boolean - checks whether the router fields are updated client-side and ready for use.
+  useEffect(() => {
+    if (!router.isReady) return;
+    setTaskNumber(router.query.task);
+    setTaskTitle(
+      data.ext_inspera_candidates[0].result.ext_inspera_questions[
+        router.query.task - 1
+      ].ext_inspera_questionTitle
+    );
+  }, [router.isReady]);
+
+  // to make sure setAssessments is being set, otherwise it is empty
+  useEffect(() => {
```

Comment:
In my editor, I get this warning:
> React Hook useEffect contains a call to 'setAssessments'. Without a list of dependencies, this can lead to an infinite chain of updates. To fix this, pass [assessments.length, p] as a second argument to the useEffect Hook.

Have you looked into it?